core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1709,6 +1709,28 @@ private[spark] object Utils extends Logging {
   }
 
   /**
+   * Terminates a process waiting for at most the specified duration. Returns whether
+   * the process terminated.
+   */
+  def terminateProcess(process: Process, timeoutMs: Long): Option[Int] = {
+    try {
+      // Java8 added a new API which will more forcibly kill the process. Use that if available.
+      val destroyMethod = process.getClass().getMethod("destroyForcibly");
+      destroyMethod.setAccessible(true)
+      destroyMethod.invoke(process)
+    } catch {
+      case e: Exception => {
+        process.destroy()
+      }
```

Comment:
this might end up swallowing useful exception messages. Maybe we should catch more narrowly
```
case NonFatal(e) =>
  if (!e.isInstanceOf[NoSuchMethodError]) {
    logWarning("Exception when attempting to kill process", e)
  }
  process.destroy()
```